core-job/src/main/java/org/apache/kylin/job/execution/AbstractExecutable.java

```diff
@@ -114,48 +147,62 @@ public final ExecuteResult execute(ExecutableContext executableContext) throws E
         logger.info("Executing AbstractExecutable (" + this.getName() + ")");
 
         Preconditions.checkArgument(executableContext instanceof DefaultContext);
-        ExecuteResult result = null;
-
-        try {
-            onExecuteStart(executableContext);
-            Throwable exception;
-            do {
-                if (retry > 0) {
-                    logger.info("Retry " + retry);
-                }
-                exception = null;
-                result = null;
-                try {
-                    result = doWork(executableContext);
-                } catch (Throwable e) {
-                    logger.error("error running Executable: " + this.toString());
-                    exception = e;
-                }
-                retry++;
-            } while (needRetry(result, exception));
 
-            //check exception in result to avoid retry on ChainedExecutable(only need retry on subtask actually)
-            if (exception != null || result.getThrowable() != null) {
-                onExecuteError(exception, executableContext);
-                throw new ExecuteException(exception);
-            }
+        onExecuteStart(executableContext);
```

Comment:
what if onExecuteStart throws exceptions? 
originally those exceptions can be caught and re-thrown as ExecuteException, now the behavior is changed.

Reply:
Good suggestion. Need an out try to include onExecuteStart